openvm/src/tmp_register_analysis.rs

```diff
@@ -0,0 +1,322 @@
+use std::collections::{BTreeMap, HashMap, HashSet};
+
+use openvm_instructions::instruction::Instruction;
+use openvm_stark_backend::p3_field::PrimeField32;
+use powdr_autoprecompiles::blocks::BasicBlock;
+
+use crate::{instruction_formatter::openvm_instruction_formatter, opcode::*, Instr};
+
+fn f_to_i64<F: PrimeField32>(f: &F) -> i64 {
+    let u = f.as_canonical_u32();
+    if u < F::ORDER_U32 / 2 {
+        u as i64
+    } else {
+        (u as i64) - (F::ORDER_U32 as i64)
+    }
+}
+
+fn possible_targets<F: PrimeField32>(
+    instruction: &Instruction<F>,
+    instruction_pc: u64,
+    pc_step: u64,
+) -> Vec<u64> {
+    let opcode = instruction.opcode.as_usize();
+
+    if opcode == OPCODE_JALR {
+        // For JALR, we don't know the target statically.
```

Comment:
You can if the previous instruction was AUIPC: https://github.com/powdr-labs/powdr/blob/main/riscv-elf/src/rv64.rs#L211